source/generators/LED450LWcone.cc

```diff
@@ -0,0 +1,160 @@
+    "Control commands of single-particle generator.");
+
+  msg_->DeclareMethod("particle", &LED450LWcone::SetParticleDefinition,
+    "Set particle to be generated.");
+
+  G4GenericMessenger::Command& min_energy =
+    msg_->DeclareProperty("min_energy", energy_min_,
+      "Set minimum kinetic energy of the particle.");
+  min_energy.SetUnitCategory("Energy");
+  min_energy.SetParameterName("min_energy", false);
+  min_energy.SetRange("min_energy>0.");
+
+  G4GenericMessenger::Command& max_energy =
+    msg_->DeclareProperty("max_energy", energy_max_,
+      "Set maximum kinetic energy of the particle");
+  max_energy.SetUnitCategory("Energy");
+  max_energy.SetParameterName("max_energy", false);
+  max_energy.SetRange("max_energy>0.");
+
+  msg_->DeclareProperty("region", region_,
+    "Set the region of the geometry where the vertex will be generated.");
+
+
+  msg_->DeclarePropertyWithUnit("momentum", "mm",  momentum_, "Set particle 3-momentum.");
+
+  msg_->DeclareProperty("min_costheta", costheta_min_,
+			"Set minimum cosTheta for the direction of the particle.");
+  msg_->DeclareProperty("max_costheta", costheta_max_,
+			"Set maximum cosTheta for the direction of the particle.");
+  msg_->DeclareProperty("min_phi", phi_min_,
+			"Set minimum phi for the direction of the particle.");
+  msg_->DeclareProperty("max_phi", phi_max_,
+			"Set maximum phi for the direction of the particle.");
+
+
+  DetectorConstruction* detconst = (DetectorConstruction*) G4RunManager::GetRunManager()->GetUserDetectorConstruction();
+  geom_ = detconst->GetGeometry();
+}
+
+
+
+LED450LWcone::~LED450LWcone()
+{
+  delete msg_;
+}
+
+
+
+void LED450LWcone::SetParticleDefinition(G4String particle_name)
+{
+  particle_definition_ =
+    G4ParticleTable::GetParticleTable()->FindParticle(particle_name);
+
+  if (!particle_definition_)
+    G4Exception("[LED450LW]", "SetParticleDefinition()",
+      FatalException, "User gave an unknown particle name.");
+}
+
+
+
+void LED450LWcone::GeneratePrimaryVertex(G4Event* event)
+{
+  // Generate uniform random energy in [E_min, E_max]
+  G4double kinetic_energy = nexus::UniformRandomInRange(energy_max_, energy_min_);
+
+    // Calculate cartesian components of momentum
+  G4double mass   = particle_definition_->GetPDGMass();
+  G4double energy = kinetic_energy + mass;
+  G4double pmod = std::sqrt(energy*energy - mass*mass);
+
+  bool fixed_momentum = momentum_ != G4ThreeVector{};
+  bool restrict_angle = costheta_min_ != -1. || costheta_max_ != 1. || phi_min_ != 0. || phi_max_ !=2.*pi;
+
+  G4ThreeVector dir; // it will be set in the if branches below
+  if (fixed_momentum) { // if the user provides a momentum direction
+    dir = momentum_.unit();
+  } else if (restrict_angle) { // if the user provides a range of angles
+    dir = RandomDirectionInRange(costheta_min_, costheta_max_, phi_min_, phi_max_);
+  }
+
+  G4ThreeVector p_dir = G4LambertianRand(dir);
+
+  G4double coseno = -1*p_dir.dot(momentum_);
```

Comment:
We tend to use English for variable names, because often they are useful to understand the code. Please, stick to that.

Reply:
Done!